src/EditorFeatures/CSharpTest/Diagnostics/UpgradeProject/UpgradeProjectTests.cs

```diff
@@ -34,6 +35,10 @@ internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProvider
                 var newSolution = appliedChanges.Item2;
                 Assert.All(newSolution.Projects.Where(p => p.Language == LanguageNames.CSharp),
                     p => Assert.Equal(expected, ((CSharpParseOptions)p.ParseOptions).SpecifiedLanguageVersion));
+
+                // Verify no document changes when upgrade project
+                var changedDocs = SolutionUtilities.GetTextChangedDocuments(oldSolution, newSolution);
+                Assert.Equal(0, changedDocs.Count());
```

Comment:
You can Assert.Empty() which gives a nicer output if the assert fails.

Reply:
fixed